File: src/api/menus.ts
import { RequestFN } from "~/core/request-function";
import { decodeWeekMenu } from "~/decoders/week-menu";
import { encodePronoteDate } from "~/encoders/pronote-date";
import { TabLocation, type WeekMenu, type SessionHandle } from "~/models";

export const menus = async (session: SessionHandle, date = new Date()): Promise<WeekMenu> => {
  const request = new RequestFN(session, "PageMenus", {
    _Signature_: { onglet: TabLocation.Menus },
    donnees: {
      date: {
        _T: 7,
        V: encodePronoteDate(date)
      }
    }
  });

  const response = await request.send();
  return decodeWeekMenu(response.data.donnees);
};

File: library/src/commonMain/kotlin/api/menus.kt
package api

import kotlinx.datetime.toKotlinLocalDateTime
import core.RequestFN
import decoders.decodeWeekMenu
import encoders.encodePronoteDate
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.*
import models.SessionHandle
import models.TabLocation
import models.WeekMenu
import java.time.LocalDateTime

suspend fun menus (session: SessionHandle, date: kotlinx.datetime.LocalDateTime = LocalDateTime.now().toKotlinLocalDateTime()): WeekMenu {
    val request = RequestFN(session.information, "PageMenus", Json.encodeToString(buildJsonObject {
        putJsonObject("_Signature_") {
            put("onglet", TabLocation.Menus.code)
        }

        putJsonObject("donnees") {
            putJsonObject("date") {
                put("_T", 7)
                put("V", encodePronoteDate(date))
            }
        }
    }))

    val response = request.send()
    return decodeWeekMenu(Json.parseToJsonElement(response.data).jsonObject["donnees"]!!.jsonObject)
}